### PR TITLE
Drop support for old Postgres versions

### PR DESCRIPTION
**Pull request by msakrejda**
In 06daccdacb25a1c8f37df472abd31c67f14df10a, we dropped CI testing for
Postgres versions older than than 9.6 (which itself has been EOL for over a
year).

Drop support in code to allow us to simplify some logic. Users who
need support for older Postgres releases can use an older collector
for the time being.

### CODE REVIEW

**Comment by msakrejda** [comment]
@keiko713 yeah, I realized we should probably do this when I saw you were accounting for 9.3 in #342. And you're right--it looks like there's some README cleanup (we mention 9.3 there). I'll do that once this gets the green light.

File: state/postgres_version.go
Comment (lfittl) [inline state/postgres_version.go:14]: There is a hard-coded "9.3 or newer is required." message on the top of `input/full.go` that also needs to be updated.

**Review by lfittl — APPROVED** [review]
:+1: Looks good, with two minor comments.

Overall, I think dropping 9.5 and older support is clearly the right choice - those versions are long gone.

9.6 and 10 there are still some late upgraders, so doesn't hurt to keep them around for a bit longer and then drop them when we either have a good reason, or more time has passed.

File: input/postgres/backends.go
Comment (msakrejda) [reply input/postgres/backends.go:38]: It's not super clear from the diff, but 9.6 is now default, so that's the `else` branch.

Comment (lfittl) [reply input/postgres/backends.go:38]: Ah, thanks for pointing that out - my bad :)

**Comment by msakrejda** [comment]
I pushed an update cleaning up other mentions of older versions. The last commit gets rid of support for parsing pre-9.6 log lines (at least those that were documented as such). Let me know if we should be more conservative and I can just drop that commit, but I checked 9.6 xlog.c and vacuumlazy.c, and those parts of the messages are always present.

File: state/postgres_statement.go
Comment (lfittl) [inline state/postgres_statement.go:41]: Should we turn these into regular `float64` values now? (or is there ever a case where they are NULL on 9.5+?)

File: state/postgres_statement.go
Comment (lfittl) [inline state/postgres_statement.go:48]: We have a fallback logic for calculating the QueryID in the statements input function (based on the query text) that we can probably drop now as well. In case we don't get a queryid a hard error is probably better than some special logic (we hash the text currently in such cases).

File: logs/analyze.go
Comment (lfittl) [inline logs/analyze.go:1618]: We can drop this logic later as well:

```
			if parts[7] != "" {
				pinskippedPages, _ := strconv.ParseInt(parts[7], 10, 64)
				logLine.Details["pinskipped_pages"] = pinskippedPages
			}
			if parts[8] != "" {
				frozenskippedPages, _ := strconv.ParseInt(parts[8], 10, 64)
				logLine.Details["frozenskipped_pages"] = frozenskippedPages
			}
```

**Review by lfittl — APPROVED** [review]
Thanks for cleaning that up!

File: logs/analyze.go
Comment (msakrejda) [reply logs/analyze.go:1618]: Right, of course.

File: state/postgres_statement.go
Comment (msakrejda) [reply state/postgres_statement.go:41]: 9.5+ should work. If someone is using a super-old un-upgraded 1.2 from Postgres 9.4, this would break. It's probably okay to drop support for that, too.

Comment (msakrejda) [reply state/postgres_statement.go:41]: I changed these to raw floats, and to my surprise, nothing broke. Do we actually [do anything with these](https://github.com/pganalyze/collector/blob/main/output/transform/postgres_statements.go#L46-L66) right now?

Comment (keiko713) [reply state/postgres_statement.go]: ... doesn't look like it's used at all? 🤔

Comment (msakrejda) [reply state/postgres_statement.go]: Got it. So we could do something like plot the latest average, but that's probably not very useful because right now it becomes more and more weighted toward the past: the number grows in inertia and will stop responding meaningfully to even dramatic changes, right?

**Comment by msakrejda** [comment]
I think this is ready to go now, but I can wait until #339 and #342 land to avoid causing unnecessary conflicts.

File: state/postgres_version.go
Comment (lfittl) [reply state/postgres_version.go:14]: Re-using this thread - On another note, I just realized we should probably raise this to Postgres 10 being the minimum - because that was the first version to use $n style replacement characters in pg_stat_statements.

See https://github.com/pganalyze/collector/pull/383, which will effectively make Postgres 9.6 not work with the collector (since we fail to parse / normalize the statements containing "?").

Comment (msakrejda) [reply state/postgres_version.go:14]: Makes sense. I'll update once #339 lands.

File: README.md
Comment (keiko713) [inline README.md:131]: There is "--- The following is needed for Postgres 10+:" above that we can remove.

**Review by lfittl — APPROVED** [review]
:+1: Re-reviewed and looks good

File: README.md
Comment (msakrejda) [reply README.md:131]: Eh, thinking about this, we presumably still need that (on all versions now) if you were to run the sequence report. Which we no longer really do. I think the README needs some TLC anyway (our docs on [pganalyze.com](https://pganalyze.com) are much more up-to-date), so I think I'll just leave as-is for now.

**Pull request merged**
